### PR TITLE
MM-54514 fix for invite_id being updated on all changes

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
The UI is sending all fields in the patch. However, the server only expects updated fields in the patch. When patch is called and `AllowOpenInvite` is set to 'false', `invite_id` gets cleared (because it the server thinks it is being set from 'true' to 'false'). It is then later recreated, because is it is empty. That fix is in `team_general_tab.tsx`. Only send the updated fields in the patch, not the entire team object.

When testing, after setting or removing the team icon, the `invite_id` still got cleared. However, only in the UX, the data was not being cleared (ie. refresh and it was back). This was due to after a patching a team, the updated team is sent to everyone via websockets. However, this team has been sanitized and therefore `invite_id` has been cleared.

I fixed this by dispatching a PATCHED_TEAM message after retrieving the patched team (see `team.ts`).

Caused during conversion to typescript -
https://github.com/mattermost/mattermost-webapp/pull/11143

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-54514

#### Release Note
```release-note
Fixed an issue where invite_id was being reset on all team changes.
```

### CODE REVIEW

File: server/channels/api4/team.go [hunk 1]
```diff
@@ -1711,10 +1722,22 @@ func removeTeamIcon(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
 	auditRec.Success()
+	auditRec.AddEventResultState(patchedTeam)
 	c.LogAudit("")
 
-	ReturnStatusOK(w)
+	if err := json.NewEncoder(w).Encode(patchedTeam); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
+	}
+
```
Comment (hanzei) [inline server/channels/api4/team.go:1740]: Let's fix CI
```suggestion
	}
```

File: server/channels/api4/team.go [hunk 2]
```diff
@@ -1685,10 +1685,21 @@ func setTeamIcon(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	c.App.SanitizeTeam(*c.AppContext.Session(), patchedTeam)
+
 	auditRec.Success()
+	auditRec.AddEventResultState(patchedTeam)
 	c.LogAudit("")
 
-	ReturnStatusOK(w)
+	if err := json.NewEncoder(w).Encode(patchedTeam); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
+	}
```
Comment (hanzei) [inline server/channels/api4/team.go:1702]: In the description, it's said:

> This was due to after a patching a team, the updated team is sent to everyone via websockets. However, this team has been sanitized and therefore `invite_id` has been cleared

Should we fix the sanitization for the websocket event instead? Should `invite_id` be present in the WS event?

**Review by hanzei — CHANGES_REQUESTED** [review]
I'm not too comfortable breaking the API endpoints. Let's check if we can fix the bug in a less invasive way.

File: server/channels/api4/team.go (hunk 2 above)
Comment (sbishel) [reply server/channels/api4/team.go]: No, it is intentionally being sanitized and should not be sent to all users.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Comment by sbishel** [comment]
@calebroseland - now ready for review...

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - this is fixed 👍

**Pull request merged**